### PR TITLE
Share selected currency state across account overview page components

### LINKED ISSUE #5753: Simplify deposits UI - Show & update the next deposit on selected currency change
Blocked until #5700 is merged.

### Description

This issue is part of the larger Epic to add support for multiple currencies in the new deposit and balances UI (https://github.com/Automattic/woocommerce-payments/issues/5750).

This issue will change the next deposit section of the deposits card to reflect changing the chosen currency on the **Payments → Overview** admin screen.

Note that we should consider if persisting the selected currency state when navigating away is appropriate.

### Acceptance criteria 

- Selecting a currency in the tab navigation should:
   - update the next deposit table to include information about the next deposit for the given currency.

### Designs

p2: paJDYF-7fi-p2

https://user-images.githubusercontent.com/8490476/224611306-583f8073-95fb-4efd-8df4-e37a4308f816.mov

### Additional context

Project: pdjTHR-2i5-p2

### LINKED ISSUE #5754: Simplify deposits UI - Update the deposit history on selected currency change
### Description

This issue is part of the larger Epic to add support for multiple currencies in the new deposit and balances UI (https://github.com/Automattic/woocommerce-payments/issues/5750).

This issue will change the next deposit section of the deposits card to reflect changing the chosen currency on the **Payments → Overview** admin screen.

### Acceptance criteria 

- Selecting a currency in the tab navigation should:
   - update the deposit history table to include the last 3 deposits for the selected currency.

### Designs

p2: paJDYF-7fi-p2

https://user-images.githubusercontent.com/8490476/224612537-c5b58831-9a7f-489e-aaab-19c440e366fa.mov

### Additional context

Project: pdjTHR-2i5-p2

### PR DESCRIPTION
**Pull request by Jinksi**
Resolves #5753
Resolves #5754 

#### Changes proposed in this Pull Request

- The selected currency state is now stored as a URL query param `selected_currency`. This allows the correct currency to remain selected when using the browser "back" button.
- This is updated when the Account Balances currency tab is selected.
- Two new hooks have been added to `clients/overview/hooks.ts`
  - `useSelectedCurrency`: for retrieving and updating the selected currency.
  - `useSelectedCurrencyOverview`: to get the selected currency overview.
- These hooks have been integrated into the Account Balances and Deposit Overview cards.

Screen recording of the state syncing between the cards and after a page refresh:

https://user-images.githubusercontent.com/3147296/231020380-ca9e157e-4ef0-4f62-bd3e-6d3c421e0c39.mov

#### Testing instructions

1. Enable the `_wcpay_feature_simplify_deposits_ui` feature flag.
To enable, run `update_option( '_wcpay_feature_simplify_deposits_ui', '1' );` in [WP Console](https://wordpress.org/plugins/wp-console/).
5. Add more than one bank account currency to your store. [See public docs](https://woocommerce.com/document/woocommerce-payments/deposits/deposit-currencies/#adding-multiple-bank-accounts).
6. Enable multi-currencies for more than one of your bank accounts.
7. Purchase some products in each currency, some using the "bypass pending" card number to add funds straight to "available funds": `4000000000000077`.
8. Select each currency in the Account Overview card and ensure the correct currency and amounts are rendered.
9. Viewing the deposit history table will require previous deposits for the selected currency. To achieve this, purchase items with the "available funds" card above, set the deposit schedule to "daily" and wait 1+ days.
10. Refreshing the page should persist the selected currency, once loading has completed.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `5822` or branch name `add/5753-overview-selected-currency-data-store` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/5753-overview-selected-currency-data-store) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: f816b7e7191ff8b51ba88a8b8a22530b2f3a089a
- Build time: 2023-04-11 02:39:03 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by Jinksi** [comment]
This PR is mostly ready but on hold, since it will be useful to have #5799 merged for testing purposes.

File: client/components/account-balances/balances-tab-panel.tsx
Comment (Jinksi) [inline client/components/account-balances/balances-tab-panel.tsx:101]: If a string is passed to initialTabName that doesn't match with a tab name, no tab will be selected. So we check to see if it's valid before passing it.

File: client/components/deposits-overview/index.tsx
Comment (Jinksi) [inline client/components/deposits-overview/index.tsx]: This new hook will return the required logic to return the deposits overview for the selected currency only.

File: client/overview/hooks.ts
Comment (Jinksi) [inline client/overview/hooks.ts:76]: Return the default currency if the selected currency is not in the overviews response.

**Comment by mattallan** [comment]
> This PR is mostly ready but on hold, since it will be useful to have https://github.com/Automattic/woocommerce-payments/pull/5799 merged for testing purposes.

I've just fixed the conflict related to some changes made to that^ PR before it was merged.

**Comment by Jinksi** [comment]
I have mentioned in the description that this PR will probably also resolve #5754
We can hold off on this review until #5700 is merged to make sure if preferred.

File: client/components/deposits-overview/test/__snapshots__/index.tsx.snap
Comment (Jinksi) [inline client/components/deposits-overview/test/__snapshots__/index.tsx.snap]: Need to investigate why this snapshot changed.

Comment (Jinksi) [reply client/components/deposits-overview/test/__snapshots__/index.tsx.snap]: Fixed in 6b51b10

File: client/components/account-balances/balances-tab-panel.tsx
Comment (Jinksi) [inline client/components/account-balances/balances-tab-panel.tsx:103]: This diff looks extensive, however all code within `<TabPanel></TabPanel>` is a copy-paste from `develop`, with indentation changes.

**Review by mattallan — APPROVED** [review]
I've been testing this PR on https://imaginative-hornet.jurassic.ninja/ and it works as expected with multiple and single currency stores.

I tried adding `&selected_currency=trash` to the overview page and confirmed it falls back to the default currency in this case.

The table row loading is a nice touch when changing between currencies and I like the hook approach used for `useSelectedCurrency`. I didn't have any feedback/suggestions on the code and since it's working as expected I'm happy to approve this one.

File: client/components/account-balances/balances-tab-panel.tsx
Comment (haszari) [reply client/components/account-balances/balances-tab-panel.tsx:103]: FYI there's a `Hide whitespace` option in the GitHub diff view, really useful for reviewing changes like this. 

In the gear menu: 

Screenshot 2023-04-12 at 9 05 22 AM

**Pull request merged**
